pkg/compute/management_client.go

```diff
@@ -128,10 +133,23 @@ func (m *ManagementClient) updateResources(ctx context.Context) {
 	}
 }
 
+func (m *ManagementClient) heartbeat(ctx context.Context, seq uint64) {
+	if err := m.heartbeatClient.SendHeartbeat(ctx, seq); err != nil {
+		log.Ctx(ctx).Error().Err(err).Msgf("heartbeat failed sending sequence %d", seq)
+	}
+}
+
 func (m *ManagementClient) Start(ctx context.Context) {
 	infoTicker := time.NewTicker(infoUpdateFrequencyMinutes * time.Minute)
 	resourceTicker := time.NewTicker(resourceUpdateFrequencySeconds * time.Second)
 
+	// The heartbeat ticker will fire twice as often as the configured, to ensure that
+	// we don't slip outside the window.  If we only ever sent on the configured
+	// frequency we are at risk of the node's liveness flapping between good and bad.
+	heartbeatTicker := time.NewTicker((heartbeatFrequencySeconds / 2) * time.Second)
+
+	var heartbeatSequence uint64 = 0
+
 	loop := true
 	for loop {
 		select {
```

Comment:
we can remove the boolean and return when ctx is done or the closeChannel is closed (maybe just call this channel `done`?) the calls to stop at the bottom of this method can then be handled via a deferred function.